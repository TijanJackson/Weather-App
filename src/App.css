
* {
  background: lightblue;
}

.App {
  text-align: center;
}


.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

#title {
  border: solid black 2px;
  padding: 10px;
  margin-top: 10px;
}

#type {
  margin-top: 40px;
  font-weight: bold;
}

#search {
  margin-top: 30px;
  
}

#weatherName {
  margin-top: 40px;
}

.btn {
  cursor: pointer;
}

#changeF {
  margin-right: 10px;
}

#changeC {
  margin-left: 10px;
}

/* #buttons {

} */
