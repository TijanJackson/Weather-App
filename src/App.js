import React, { Component } from 'react';
import 'isomorphic-fetch'; 
import './App.css';



class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            error: null,
            main: '',
            mainMin: '',
            name: 'City name',
            celsius: true,
            fahrenheit: false,
            kelvin: 273,
            multiply: 1,
            thirtyTwo: 0
        };
        this.searchWeather = this.searchWeather.bind(this);
        this.changeToF = this.changeToF.bind(this);
        this.changeToC = this.changeToC.bind(this);
    }

    // converts to fahrenheit
    changeToF(e) {
        this.setState({
            celsius: false,
            fahrenheit: true,
            multiply: 1.8,
            thirtyTwo: 32

        })
    }

    // converts to celsius
    changeToC(e) {
        this.setState({
            celsius: true,
            fahrenheit: false,
            multiply: 1,
            thirtyTwo: 0

        })
    }

   // http://api.openweathermap.org/data/2.5/weather?q=london&APPID=94033fe10a92947fef19fe1f5c0506c7

    searchWeather() {
        fetch(`http://api.openweathermap.org/data/2.5/weather?q=${document.getElementById('weather').value}&APPID=94033fe10a92947fef19fe1f5c0506c7`)
            .then(res => res.json()) 
            .then(
                (result) => {
                    this.setState({
                        main: result.main.temp_max,
                        mainMin: result.main.temp_min,
                        name: result.name
                    });
                },
                // error handling takes place here
                (error) => {
                    this.setState({
                        error
                    });
                })
    }

    render() {
     const { error,  main, mainMin, multiply, thirtyTwo, name} = this.state;
     if (error) {
         return <div>Error: {error.message}</div>;
     } 
    else {
         return (
             <div className="App">
               <h1 id="title">Weather App</h1>
               <form>
                   <div id="type">
                   <label>Type a City Name: </label>
                   <input id="weather" type="text" placeholder=" Type a city name here"></input>
                   </div>
               </form>
               <div id="buttons">
               <button className="btn" id="search" onClick={this.searchWeather}>Check temperature</button>
               <br></br>
               <br></br>
               <button className="btn" id="changeF" onClick={this.changeToF}>Change to fahrenheit</button>
               <button className="btn" id="changeC" onClick={this.changeToC}>Change to celsius</button>
               </div>
               {/* Displays the initial value, then after the search function has run, it displays
               the value retrieved and subtracts 273.15 from it in order to display the weather in celsius */}
               <h2 id="weatherName">{name}</h2>
               <div id="weatherDetails">
               <h3>Max temp: {Math.round( ((main - 273.15) * multiply) + thirtyTwo)}°C</h3>
               <h3>Min Temp: {Math.round( ((mainMin - 273.15) * multiply) + thirtyTwo)}°C</h3>
               </div>
            </div>
         );
     }
 }
 }


 export default App;