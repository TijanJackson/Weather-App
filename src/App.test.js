import React from 'react';
// import { render } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';


// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });


// test('the data is peanut butter', async () => {
//   const data = await fetchData();
//   expect(data).toBe('peanut butter');
// });

// test('the fetch fails with an error', async () => {
//   expect.assertions(1);
//   try {
//     await fetchData();
//   } catch (e) {
//     expect(e).toMatch('error');
//   }
// });



// test('the data is peanut butter', () => {
//   return fetchData().then(data => {
//     expect(data).toBe('peanut butter');
//   });
// });

// test('renders correctly', () => {
//   const tree = renderer
//   .create(<App></App>)
//   .toJSON();
//   expect(tree).toMatchSnapshot();
//   });
  

// test('renders correctly', () => {
//   const tree = renderer
//   .create(<App></App>)
//   .toJSON();
//   expect(tree).toMatchSnapshot();
//   });
  

// test('the data should be the weather', () => {
//   return fetch().then(data => {
//     expect(data).toBe('weather');
//   });
// });

// test('the data is weather', () => {
//   return expect(fetchData()).resolves.toBe('weather');
// });

// test('adds 1 + 2 to equal 3', () => {
//   expect(sum(1, 2)).toBe(3);
// });

// function searchWeather() {
//   fetch(`http://api.openweathermap.org/data/2.5/weather?q=${document.getElementById('weather').value}&APPID=94033fe10a92947fef19fe1f5c0506c7`)
//       .then(res => res.json()) 
//       .then(
//           (result) => {
//               this.setState({
//                   main: result.main
//               });
//           },
//           // error handling takes place here
//           (error) => {
//               this.setState({
//                   error
//               });
//           })
// }

//   fetch(`http://api.openweathermap.org/data/2.5/weather?q=London&APPID=94033fe10a92947fef19fe1f5c0506c7`)
//       .then(res => res.json()) 
//       .then(
//           (result) => {
//               this.setState({
//                   main: result.main
//               });
//           },
//           // error handling takes place here
//           (error) => {
//               this.setState({
//                   error
//               });
//           })


test('fetches data', () => {
  expect(fetch(`http://api.openweathermap.org/data/2.5/weather?q=London&APPID=94033fe10a92947fef19fe1f5c0506c7`)).toBe({"coord":{"lon":-0.13,"lat":51.51},"weather":[{"id":804,"main":"Clouds","description":"overcast clouds","icon":"04d"}],"base":"stations","main":{"temp":286.97,"feels_like":281.5,"temp_min":284.82,"temp_max":288.71,"pressure":1004,"humidity":82},"visibility":10000,"wind":{"speed":8.2,"deg":240},"clouds":{"all":90},"dt":1583845089,"sys":{"type":1,"id":1414,"country":"GB","sunrise":1583821515,"sunset":1583862983},"timezone":0,"id":2643743,"name":"London","cod":200});
});

