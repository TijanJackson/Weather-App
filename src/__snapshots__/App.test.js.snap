// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    Weather App
  </h1>
  <form>
    <label>
      Type a City Name: 
    </label>
    <input
      id="weather"
      type="text"
    />
  </form>
  <button
    onClick={[Function]}
  >
    Check temperature
  </button>
  <h3>
    Temp: 
    NaN
    °C
  </h3>
  <h3>
    Min Temp: 
    NaN
    °C
  </h3>
</div>
`;
